infra/terraform/README.md

```diff
@@ -0,0 +1,249 @@
+Certainly! Here’s the updated `README.md` with the new section added for setting up AWS roles and permissions:
+
+---
+
+# OWTF Terraform Scripts
+
+## Overview of the Bash Script
+
+This script automates the process of modifying a Terraform configuration file, checking for unsafe modifications, and running Terraform commands. It also handles different operating systems (macOS and Linux) and allows the user to choose whether to run the script in a "safe mode" or "unsafe mode."
+
+## Deployment Steps
+
+1. **Clone the Repository**
+
+    First, clone the repository containing the deployment script and Kubernetes manifests:
+    ```bash
+    git clone https://github.com/owtf/owtf.git
+
+    cd owtf/infra/terraform
+    ```
+
+2. **Install Terraform and AWS CLI**
+
+    Make sure to install [Terraform](https://developer.hashicorp.com/terraform/install) and [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html).
+
+3. **Setting Up AWS Roles and Permissions**
+
+      To ensure secure and efficient use of Terraform with AWS, you need to set up AWS IAM roles and permissions correctly. This section provides instructions for setting up roles with either Single Sign-On (SSO) or user-based role assumption.
+
+      ### Using Single Sign-On (SSO)
+
+   AWS Single Sign-On (SSO) allows you to manage access to your AWS resources using your organization's SSO provider. Here's how to set it up:
+
+   1. **Configure AWS SSO:**
+      - Sign in to the [AWS SSO Console](https://console.aws.amazon.com/singlesignon/).
+      - Follow the [AWS documentation](https://docs.aws.amazon.com/singlesignon/latest/userguide/getting-started.html) to set up SSO with your identity provider.
+
+   2. **Create IAM Roles for SSO:**
+      - Go to the [IAM Console](https://console.aws.amazon.com/iam/home).
+      - Create a new role and select **AWS SSO** as the trusted entity.
+      - Assign permissions that are necessary for Terraform operations. Common permissions include `AmazonEC2FullAccess`, `AmazonVPCFullAccess`, `AmazonS3FullAccess`, and `IAMFullAccess`.
```

Comment:
Can you modify this to include exactly the set of permissions required for the script? This seems very GPT generalized explanation to me.